documentation/client-side-forms/appendix-vaadin-components.asciidoc

```diff
@@ -0,0 +1,284 @@
+
+[source, typescript]
+----
+import '@vaadin/vaadin-checkbox';
+import '@vaadin/vaadin-radio-button';
+...
+static get styles() {
+  return css`
+    vaadin-checkbox[invalid], vaadin-radio-button[invalid],
+      background: var(--lumo-error-color-10pct);
+    }`;
+}
+...
+render() {
+  return html`
+  <vaadin-checkbox ...="${field(this.binder.model.myBooleanField)}">checkbox</vaadin-checkbox>
+  <vaadin-radio-button ...="${field(this.binder.model.myBooleanField)}">radio-button</vaadin-radio-button>
+  `;
+}
+----
+
+=== Binding data to list fields
+
+Vaadin has several components for selection based on option lists, each one covering a specific purpose,
+hence the way to set their values and options differs.
+
+==== Configuring the options for selection
+
+Options for these components can be set by calling a server side service that provides the list of strings.
+Since the call to the endpoint is asynchronous, one easy way is to combine the `until` and `repeat` methods in the `lit-html` library.
+
+As reference, the following snippet demonstrates how to repeat a pattern given an asynchronous method that returns a list of items. The same pattern will be used in the code blocks for each component below.
+
+[source, typescript]
+----
+import {until} from 'lit-html/directives/until.js';
+import {repeat} from 'lit-html/directives/repeat.js';
+...
+render() {
+  return html`
+  ...
+    ${until(endpoint.getMyOptions().then(opts => repeat(opts, (item) => html`
+      <div>${item}</div>
+    `)))}
+  ...
+  `;
+}
+----
+
+==== Single selection using the item value
+
+For a single selection `vaadin-combo-box`, `vaadin-radio-group` or `vaadin-list-box` should be used.
+All of them can take the selected item value as a string.
+
+[source, typescript]
+----
+import '@vaadin/vaadin-combo-box';
+import '@vaadin/vaadin-list-box';
+import '@vaadin/vaadin-radio-button/vaadin-radio-group';
+...
+render() {
+  return html`
+  <vaadin-combo-box ...="${field(this.binder.model.mySingleSelectionField)}"
+    .items="${until(endpoint.getMyOptions())}" label="combo-box">
+  </vaadin-combo-box>
+
+  <vaadin-radio-group ...="${field(this.binder.model.mySingleSelectionField)}" label="radio-group">
+    ${until(endpoint.getMyOptions().then(opts => repeat(opts, (item) => html`
+      <vaadin-radio-button value="${item}">${item}</vaadin-radio-button>
+    `)))}
+  </vaadin-radio-group>
+
+  <vaadin-list-box ...="${field(this.binder.model.mySingleSelectionField)}" label="list-box">
+    ${until(endpoint.getMyOptions().then(opts => repeat(opts, (item) => html`
+      <vaadin-item><span>${item}</span></vaadin-item>
+    `)))}
+  </vaadin-list-box>
+  `;
+}
+----
+
+==== Single selection using index
+
+To select items by index, the `vaadin-select` component should be used. It accepts an integer for the index value.
+Because this component is configurable via the `template` tag, it's not possible to set the options with an asynchronous call.
```

Comment:
I think this is a bit wrong. `vaadin-select` can also be configured with a `renderer` function instead of `<template>` but I'm pretty sure you can also set the whole template asynchronously. It should observe changes to template (when the whole template element is changed) and renderer. Though I'm not sure if you would need to make sure to set the selected value only after the template has been set.

Reply:
Also I'm not sure what is the best case to show in this example (template vs renderer or how to structure the example).

Reply:
Here's an example using TS and `vaadin-select` with `renderer`: https://vaadin-ts-examples.herokuapp.com/select-renderer